src/Microsoft.ML.Core/Utilities/DoubleParser.cs

```diff
@@ -85,24 +102,24 @@ public enum Result
         /// <summary>
         /// This produces zero for an empty string.
         /// </summary>
-        public static bool TryParse(ReadOnlySpan<char> span, out Single value)
+        public static bool TryParse(ReadOnlySpan<char> span, out Single value, char decimalMarker = Defaults.DecimalMarker)
```

Comment:
Originally I thought in actually adding a `DoubleParser.Options options` parameter instead of the `char decimalMarker` one, but then I couldn't set a default options object, because the compiler would complain that it needs to have this parameter at compile time. So I think the best solution is to add new parameters for new DoubleParser options, and just let the callers of these to use DoubleParser.Options to get the values for the parameters.

Reply:
Now that I've refactored DoubleParser.Options into an enum, it is possible to set a default in here, and I've done that.

---
In reply to: [429522604](https://github.com/dotnet/machinelearning/pull/5154#discussion_r429522604) [](ancestors = 429522604)